src/tests/conformance/mod.rs

```diff
@@ -1,3 +1,36 @@
+use std::io;
+
+use crate::{
+    setup::node::{Action, Node},
+    tools::synthetic_node::SyntheticNode,
+};
+
 mod handshake;
 mod invalid_message;
 mod messages;
+mod unsolicited_response;
+
+/// Creates a connected [Node] and [SyntheticNode] which have completed the full handshake.
+///
+/// The [SyntheticNode] initiated the connection.
+///
+/// The [SyntheticNode]'s MessageFilter has been configured for auto replying.
+async fn simple_handshaken_node() -> io::Result<(Node, SyntheticNode)> {
```

Comment:
```suggestion
async fn simple_handshaken_node_and_synth_node() -> io::Result<(Node, SyntheticNode)> {
```

Reply:
The downside to this is we loose the explictness of the setup in the test cases 🤔

Reply:
I don't mind removing this again.

Naming could definitely be better. Some other options:

1. `setup_basic_handshaken_connection`
2. `node_with_single_handshaken_connection`

Reply:
I think it might be best to keep the cases flexible albeit a little less DRY for now since the suite is still fairly young and will likely change in future?

Reply:
Sure, I can just move it up one notch back into the same file at least.